cdap-spark-core/src/main/scala/co/cask/cdap/app/runtime/spark/DefaultJavaSparkExecutionContext.scala

```diff
@@ -83,12 +83,29 @@ class DefaultJavaSparkExecutionContext(sec: SparkExecutionContext) extends JavaS
       sec.fromDataset(SparkRuntimeEnv.getContext, datasetName, arguments.toMap, Option(splits).map(_.toIterable)))
   }
 
-  override def fromStream(streamName: String, startTime: Long, endTime: Long) : JavaRDD[StreamEvent] = {
+  override def fromDataset[K, V](datasetNamespace: String, datasetName: String, arguments: util.Map[String, String],
+                                 splits: java.lang.Iterable[_ <: Split]): JavaPairRDD[K, V] = {
+    // Create the implicit fake ClassTags to satisfy scala type system at compilation time.
+    implicit val kTag: ClassTag[K] = createClassTag
+    implicit val vTag: ClassTag[V] = createClassTag
+    JavaPairRDD.fromRDD(
+      sec.fromDataset(SparkRuntimeEnv.getContext, datasetNamespace, datasetName, arguments.toMap,
+        Option(splits).map(_.toIterable)))
+  }
+
+  override def fromStream(streamName: String, startTime: Long, endTime: Long): JavaRDD[StreamEvent] = {
     val ct: ClassTag[StreamEvent] = createClassTag
     JavaRDD.fromRDD(
       sec.fromStream(SparkRuntimeEnv.getContext, streamName, startTime, endTime)(ct, (e: StreamEvent) => e))
   }
 
+  override def fromStream(namespace: String, streamName: String, startTime: Long, endTime: Long):
```

Comment:
nit: the parameter is called 'namespace' for the stream methods and ``datasetNamespace`` for dataset methods?  It should be consistent